containers/datalab/content/GCSbackup.sh

```diff
@@ -96,10 +96,26 @@ timestamp=$(date "+%Y%m%d%H%M%S")
 project_id=${project_id:-$VM_PROJECT}
 zone=${zone:-${VM_ZONE}}
 machine_name=${machine_name:-$VM_NAME}
-default_bucket="${project_id}.appspot.com"
 tag="${tag:-backup}"
 num_backups=${num_backups:-10}
-gcs_bucket=${gcs_bucket:-$default_bucket}
+
+# If no bucket is provided, try $project_id.appspot.com, then try $project_id
+if [ -z "${gcs_bucket}" ]; then
+  default_bucket="${project_id}.appspot.com"
+  echo "Will use ${default_bucket} for bucket name"
+  gsutil ls "gs://${default_bucket}" &>/dev/null && gcs_bucket="${default_bucket}" || {
+    # We cannot create the $project_id.appspot.com bucket, so don't try that
+    gcs_bucket=${project_id}
+    echo "Could not list bucket ${default_bucket}. Next will use ${gcs_bucket} for bucket name"
+    gsutil mb gs://"${gcs_bucket}" || true
+  }
+else
+  gsutil ls gs://"${gcs_bucket}" &>/dev/null || {
+    echo "Could not list bucket '${gcs_bucket}'. Will try to create it.."
+    gsutil mb gs://"${gcs_bucket}" || true
```

Comment:
Should this have an "|| true"?

If this mb call fails, then the script should exit.